libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -128,10 +221,9 @@ fun DirectionsRoute.toNavigationRoute(): NavigationRoute {
     val routes = Array(routeIndex + 1) { arrIndex ->
         if (arrIndex != routeIndex) {
             // build a fake route to fill up the index gap in the response
-            DirectionsRoute.builder()
+            fakeDirectionsRoute
+                .toBuilder()
                 .routeIndex(arrIndex.toString())
```

Comment:
Have a question which is our of scope for this PR? Just want to know how the SDK works. Why do we do this? Does NN treat fake routes as alternatives?

Reply:
No, however, we need the fake routes to maintain the correct index of the actual route in the `DirectionsResponse` that this extensions tries to recreate.

This is needed, so that calling `NavigationRoute#directionsResponse#routes()[routeIndex]` always returns a correct value. Without padding the response with fake routes, the `DirectionsRoute` that we're wrapping would be at a wrong index in the response.